src/screen/dashboard/sidebar.rs

```diff
@@ -61,39 +60,176 @@ impl Sidebar {
     pub fn new() -> Self {
         Self {
             hidden: false,
-            timestamp: None,
+            reloading_config: false,
         }
     }
 
     pub fn toggle_visibility(&mut self) {
         self.hidden = !self.hidden
     }
 
-    pub fn update(&mut self, message: Message) -> Event {
+    pub fn update(&mut self, message: Message) -> Option<Event> {
         match message {
-            Message::Open(source) => Event::Open(source),
-            Message::Popout(source) => Event::Popout(source),
-            Message::Focus(window, pane) => Event::Focus(window, pane),
-            Message::Replace(window, source, pane) => Event::Replace(window, source, pane),
-            Message::Close(window, pane) => Event::Close(window, pane),
+            Message::Open(source) => Some(Event::Open(source)),
+            Message::Popout(source) => Some(Event::Popout(source)),
+            Message::Focus(window, pane) => Some(Event::Focus(window, pane)),
+            Message::Replace(window, source, pane) => Some(Event::Replace(window, source, pane)),
+            Message::Close(window, pane) => Some(Event::Close(window, pane)),
             Message::Swap(from_window, from_pane, to_window, to_pane) => {
-                Event::Swap(from_window, from_pane, to_window, to_pane)
+                Some(Event::Swap(from_window, from_pane, to_window, to_pane))
             }
-            Message::Leave(buffer) => Event::Leave(buffer),
-            Message::ToggleFileTransfers => Event::ToggleFileTransfers,
-            Message::ToggleCommandBar => Event::ToggleCommandBar,
-            Message::ToggleThemeEditor => Event::ToggleThemeEditor,
+            Message::Leave(buffer) => Some(Event::Leave(buffer)),
+            Message::ToggleFileTransfers => Some(Event::ToggleFileTransfers),
+            Message::ToggleCommandBar => Some(Event::ToggleCommandBar),
+            Message::ToggleThemeEditor => Some(Event::ToggleThemeEditor),
             Message::ReloadConfigFile => {
-                self.timestamp = Some(Instant::now());
-                Event::ReloadConfigFile
+                self.reloading_config = true;
+                Some(Event::ReloadConfigFile)
+            },
+            Message::OpenReleaseWebsite => Some(Event::OpenReleaseWebsite),
+            Message::ReloadComplete => {
+                self.reloading_config = false;
+                None
             }
-            Message::OpenReleaseWebsite => Event::OpenReleaseWebsite,
         }
     }
 
+fn menu_buttons<'a>(
+    &'a self,
+    main_window: window::Id,
+    panes: &Panes,
+    config: data::config::Sidebar,
+    show_tooltips: bool,
+    file_transfers: &'a file_transfer::Manager,
+    version: &'a Version,
+    theme_editor_open: bool,
+) -> Element<'a, Message> {
```

Comment:
Nit: Just pass `reloading_config` as a param instead of making this a method of `Sidebar`. This will also prevent this entire code-block from showing up as `+` diff which makes it very hard to review for other changes.

Reply:
I was thinking that this function would be better off as a method that takes &self instead of a reference to a few of its fields

but that can be a different PR (it definitely crowds up this one)

Reply:
Yeah that's definitely fair. Another thing that can be refactored within `menu_buttons` is we can have a button helper closure / function since all the various buttons are basically the same shape with different data.